ssl-config/src/test/java/com/palantir/ssl/SslSocketFactoriesConnectionTests.java

```diff
@@ -0,0 +1,482 @@
+    public void testSslWithClientAuthenticationJks() {
+        TrustStoreConfiguration serverTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration serverKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.SERVER_KEY_STORE_JKS_PATH)
+                .password(TestConstants.SERVER_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trust(serverTrustStoreConfig)
+                .key(serverKeyStoreConfig)
+                .build();
+
+        TrustStoreConfiguration clientTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration clientKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.CLIENT_KEY_STORE_JKS_PATH)
+                .password(TestConstants.CLIENT_KEY_STORE_JKS_PASSWORD)
+                .build();
+
+        SslConfiguration clientConfig = SslConfiguration.builder()
+                .trust(clientTrustStoreConfig)
+                .key(clientKeyStoreConfig)
+                .build();
+
+        runSslConnectionTest(serverConfig, clientConfig, ClientAuth.WITH_CLIENT_AUTH);
+    }
+
+    @Test
+    public void testSslWithClientAuthenticationPkcs12() {
+        TrustStoreConfiguration serverTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration serverKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.SERVER_KEY_STORE_P12_PATH)
+                .type(TestConstants.SERVER_KEY_STORE_P12_TYPE)
+                .password(TestConstants.SERVER_KEY_STORE_P12_PASSWORD)
+                .build();
+
+        SslConfiguration serverConfig = SslConfiguration.builder()
+                .trust(serverTrustStoreConfig)
+                .key(serverKeyStoreConfig)
+                .build();
+
+        TrustStoreConfiguration clientTrustStoreConfig = TrustStoreConfiguration.builder()
+                .path(TestConstants.CA_TRUST_STORE_PATH)
+                .build();
+
+        KeyStoreConfiguration clientKeyStoreConfig = KeyStoreConfiguration.builder()
+                .path(TestConstants.CLIENT_KEY_STORE_P12_PATH)
+                .type(TestConstants.CLIENT_KEY_STORE_P12_TYPE)
+                .password(TestConstants.CLIENT_KEY_STORE_P12_PASSWORD)
+                .build();
+
+        SslConfiguration clientConfig = SslConfiguration.builder()
+                .trust(clientTrustStoreConfig)
+                .key(clientKeyStoreConfig)
+                .build();
+
+        runSslConnectionTest(serverConfig, clientConfig, ClientAuth.WITH_CLIENT_AUTH);
+    }
+
+    @Test
+    public void testSslWithClientAuthenticationMixed() {
```

Comment:
Sure, I'll add that as a single working test case. Wanted to just use sockets for the bulk of these because they're much lighter-weight to configure (don't have to have a separate test file/test rule for every different server configuration I would want to set up), but makes sense to have at least one working test case for illustrative purposes for Dropwizard.